app/src/main/java/br/com/dionataferraz/vendas/transaction/TransactionsActivity.kt

```diff
@@ -1,77 +1,103 @@
-import br.com.dionataferraz.vendas.transaction.data.TransactionEntity
-import com.squareup.moshi.JsonAdapter
-import com.squareup.moshi.Moshi
-import com.squareup.moshi.Types
-import com.squareup.moshi.adapters.Rfc3339DateJsonAdapter
-import com.squareup.moshi.kotlin.reflect.KotlinJsonAdapterFactory
-import java.util.Date
+import br.com.dionataferraz.vendas.login.data.local.VendasDatabase
+import br.com.dionataferraz.vendas.transaction.local.TransactionDatabase
+import br.com.dionataferraz.vendas.transaction.local.TransactionEntity
+import kotlin.streams.toList
+
 
 class TransactionsActivity : AppCompatActivity(), TransactionAdapter.Listener {
 
     private lateinit var binding: ActivityTransactionsBinding
+    private lateinit var viewModel: TransactionViewModel
     private val adapter: TransactionAdapter by lazy {
         TransactionAdapter(this)
     }
 
-    private val databaseTransac: TransactionDatabase by lazy {
-        TransactionDatabase.getInstance(context = App.context)
-    }
-
     @RequiresApi(Build.VERSION_CODES.O)
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         binding = ActivityTransactionsBinding.inflate(layoutInflater)
         setContentView(binding.root)
         configureActionBar()
 
-        binding.rcList.adapter = adapter
-
-        adapter.addList(getTransactionList())
-//        setupTransactionList()?.let { adapter.addList(it) }
-    }
-
-    @RequiresApi(Build.VERSION_CODES.O)
-    private fun setupTransactionList(): List<TransactionEntity>? {
-        val sharedPreferences = getSharedPreferences("transacListPrefs", Context.MODE_PRIVATE);
-        var jsonTransactions = sharedPreferences.getString("transactionList", "");
-        val jsonAdapter = getJsonTransactionListAdapter();
-
-        if (jsonTransactions.isNullOrBlank()) {
-            val listJson = jsonAdapter?.toJson(getTransactionList())
-            sharedPreferences
-                .edit()
-                .putString("transactionList", listJson)
-                .apply()
-
-            jsonTransactions = sharedPreferences.getString("transactionList", "")
+        viewModel = TransactionViewModel()
+        getProgressBar().visibility = View.VISIBLE
+        viewModel.updateTrasactionView()
+
+        val itemTouchHelper = ItemTouchHelper(getSwipe())
+        itemTouchHelper.attachToRecyclerView(binding.rcList)
+
+        viewModel.transactionList.observe(this) { transactions ->
+            if (transactions != null && !transactions.isEmpty()) {
+                val transactionsE = transactions.stream().map {
+                    TransactionEntity(
+                        id = it.id,
+                        amount = it.value,
+                        name = it.description,
+                        time = it.transactionDate,
+                        transactionType = it.transactionType
+                    )
+                }.toList()
```

Comment:
Nesse caso só o map já serviria
```suggestion
                val transactionsE = transactions.map {
                    TransactionEntity(
                        id = it.id,
                        amount = it.value,
                        name = it.description,
                        time = it.transactionDate,
                        transactionType = it.transactionType
                    )
                }
```